pdcupdater/handlers/depchain/rpms.py

```diff
@@ -0,0 +1,249 @@
+    #stable_tags = [r['stable_tag'] for r in releases]
+    #return stable_tags + [rawhide_tag()]
+    return [rawhide_tag()]
+
+
+class BaseRPMDepChainHandler(pdcupdater.handlers.BaseHandler):
+    """ Abstract base class. """
+
+    def __init__(self, *args, **kwargs):
+        super(BaseRPMDepChainHandler, self).__init__(*args, **kwargs)
+        self.koji_url = self.config['pdcupdater.koji_url']
+
+    @property
+    def topic_suffixes(self):
+        return ['buildsys.tag']
+
+    def can_handle(self, msg):
+        if not msg['topic'].endswith('buildsys.tag'):
+            return False
+
+        # Ignore secondary arches for now
+        if msg['msg']['instance'] != 'primary':
+            log.debug("From %r.  Skipping." % (msg['msg']['instance']))
+            return False
+
+        interesting = interesting_tags()
+        tag = msg['msg']['tag']
+
+        if tag not in interesting:
+            log.debug("%r not in %r.  Skipping."  % (tag, interesting))
+            return False
+
+        return True
+
+    def _yield_pdc_relationships(self, pdc, release_id):
+        for type in self.managed_types:
+            entries = pdc.get_paged(
+                pdc['release-component-relationships']._,
+                from_component_release=release_id,
+                type=type,
+            )
+            for entry in entries:
+                yield entry
+
+    def _yield_koji_relationships(self, pdc, tag):
+
+        release_id, release = tag2release(tag)
+        # TODO -- this tag <-> release agreement is going to break down with modularity.
+
+        pdcupdater.utils.ensure_release_exists(pdc, release_id, release)
+
+        builds = pdcupdater.services.koji_builds_in_tag(self.koji_url, tag)
+
+        for build in builds:
+            parent = {
+                'name': build['name'],
+                'release': {'release_id': release_id},
+            }
+
+            relationships = self.get_koji_relationships_from_build(
+                self.koji_url, build['build_id'])
+            relationships = list(relationships)
+
+            for relationship, child in relationships:
+                child = {'name': child, 'release': {'release_id': release_id}}
+                yield parent, relationship, child
+
+    def handle(self, pdc, msg):
+        tag = msg['msg']['tag']
+        release_id, release = tag2release(tag)
+
+        # TODO -- this tag <-> release agreement is going to break down with modularity.
+        pdcupdater.utils.ensure_release_exists(pdc, release_id, release)
+
+        name = msg['msg']['name']
+        parent = pdcupdater.utils.ensure_release_component_exists(
+            pdc, release_id, name)
+
+        # Go to sleep due to a race condition that is koji's fault.
+        # It publishes a fedmsg message before the task is actually done and
+        # committed to their database.  In the next step, we try to query
+        # them -- but if the task isn't done, we get an exception.
+        time.sleep(1)
```

Comment:
Oh okay, just link to the PR then in a TODO comment. That should be good enough.